### PR TITLE
Add a monolithic/god function anti-pattern

### PR DESCRIPTION
**Pull request by schroedk**

### CODE REVIEW

**Review by mdbenito — COMMENTED** [review]
Left a couple of comments

File: anti-patterns/monolithic-function/README.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+1. Loads the data by handling different file formats.
+2. Cleans the data by filling in missing values.
+3. Normalizes the data through scaling.
+4. Performs feature engineering by creating interaction terms.
+5. Trains a machine learning model.
+6. Evaluates model performance.
+
+While this function does manage to accomplish several tasks, it suffers from poor readability. The mixture of different responsibilities within a single function makes it difficult to follow what’s happening at a glance.
+
+Additionally, the function is not easily testable and hard to modify or extend. Because it handles everything from data loading to model evaluation, testing individual parts of the process in isolation is nearly impossible.
```
[the remark this answers is not in the document]
Comment (schroedk) [reply anti-patterns/monolithic-function/README.md:74]: I would like to keep the scope very narrow here. More correct would be "imagine in order to test the data reading and preprocessing you have to train a model". This is more detail than needed, I think. What is your opinion?

File: anti-patterns/monolithic-function/README.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+def train_model(X_train: pd.DataFrame, y_train: pd.DataFrame) -> DecisionTreeClassifier:
+    model = DecisionTreeClassifier()
+    model.fit(X_train, y_train)
+    return model
+
+
+def evaluate_model(model: DecisionTreeClassifier, X_test: pd.DataFrame, y_test: pd.DataFrame) -> float:
+    y_pred = model.predict(X_test)
+    accuracy = accuracy_score(y_test, y_pred)
+    print(f"Model Accuracy: {accuracy:.2f}")
```
[the remark this answers is not in the document]
Comment (schroedk) [reply anti-patterns/monolithic-function/README.md:150]: Again, I don't want to speak about pure functions at all. In the end the print is not needed to make the message, I would remove it completely.

File: anti-patterns/monolithic-function/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mdbenito) [reply anti-patterns/monolithic-function/README.md:73]: I agree. I just thought that the print statement was a good example of a preposterous situation. But maybe too much so.

File: anti-patterns/monolithic-function/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mdbenito) [reply anti-patterns/monolithic-function/README.md]: Sorry, I didn't mean to mention anything about purity in the text. It was meant for you only: "evaluate" should have no side effects IMO, and there should be a distinct "report" step.

**Pull request merged**